File: 3/zad1/zad1.1/tiger.c
#include <stdlib.h> 

typedef char const* (*PTRFUN)();

typedef struct {
	PTRFUN* vtable;
	char const* name;
} Tiger;

char const* name(Tiger* t) { return t->name; }
char const* greet(void) { return "rawr!"; }
char const* menu(void) { return "divljac"; }

PTRFUN vtable[] = {(PTRFUN)name, greet, menu};

void* create(char const* name) {
	Tiger* t = (Tiger*)malloc(sizeof(Tiger));
	t->vtable = vtable;
	t->name = name;
	return t;
}

void* createStack(char const* name) {
	Tiger t;
	t.vtable = vtable;
	t.name = name;
	return &t;
}

File: 3/zad1/zad1.4/src/main/java/hr/fer/zemris/ooup/lab3/zad1/model/plugins/Tiger.java
package hr.fer.zemris.ooup.lab3.zad1.model.plugins;

import hr.fer.zemris.ooup.lab3.zad1.model.Animal;

public class Tiger extends Animal {

    private final String name;

    public Tiger(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public String greet() {
        return "rawr!";
    }

    @Override
    public String menu() {
        return "divljac";
    }

}
